### PR TITLE
fix(kit): correct detect iso code for `Kazakhstan` when phone number start with `+7`

### LINKED ISSUE #4193: 🐞 - InputPhoneInternational changes RU flag on KZ in case with +7 code
### Playground Link

https://taiga-ui.dev/components/input-phone-international

### Description

How to reproduce:
1. In case with all countries sorted by alphabet choose Russia
2. Paste number with +7 code in the input
3. ... flag was changed on KZ for no reason

How to solve:
Choose KZ flag in case when number starts with +7 (600—799) or +997. In other cases choose RU flag ([wikipedia](https://ru.wikipedia.org/wiki/%D0%A2%D0%B5%D0%BB%D0%B5%D1%84%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9_%D0%BF%D0%BB%D0%B0%D0%BD_%D0%BD%D1%83%D0%BC%D0%B5%D1%80%D0%B0%D1%86%D0%B8%D0%B8_%D0%9A%D0%B0%D0%B7%D0%B0%D1%85%D1%81%D1%82%D0%B0%D0%BD%D0%B0))

### Angular version

Any

### Taiga UI version

Any

### Which browsers have you used?

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #4193

### CODE REVIEW

File: projects/kit/components/input-phone-international/input-phone-international.component.ts [hunk 1]
```diff
@@ -246,13 +246,22 @@ export class TuiInputPhoneInternationalComponent
...
                 ruCodeTest ||
                 (value.startsWith(
                     this.isoToCountryCode(countryIsoCode).replace(CHAR_PLUS, ''),
                 ) &&
-                    value.length + 1 === this.getMaxAllowedLength(countryIsoCode))
-            );
+                    value.length + 1 === this.getMaxAllowedLength(countryIsoCode));
+
+            return matched && countryIsoCode === TuiCountryIsoCode.RU
+                ? this.ruCodeShouldNotBeIncludeKzRegion(value)
```
Comment (waterplea) [inline projects/kit/components/input-phone-international/input-phone-international.component.ts:257]: 1. Let's not make long names (`notKzRegion` should be ok)
2. Let's move it to a function so the name is minified.

File: projects/kit/components/input-phone-international/test/input-phone-international.component.spec.ts [hunk 2]
```diff
@@ -141,6 +141,24 @@ describe(`InputPhoneInternational`, () => {
             expect(component.countryIsoCode).toBe(TuiCountryIsoCode.RU);
         });
 
+        describe(`should be set KZ country code on paste event`, () => {
```
Comment (waterplea) [inline projects/kit/components/input-phone-international/test/input-phone-international.component.spec.ts:144]: `should set`

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-phone-international/test/input-phone-international.component.spec.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/components/input-phone-international/test/input-phone-international.component.spec.ts]: fixed

File: projects/kit/components/input-phone-international/input-phone-international.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**